src/protocol/modulus_conversion/convert_shares.rs

```diff
@@ -90,16 +90,36 @@ pub fn convert_bit_local<F: Field, B: BitArray>(
 pub fn convert_all_bits_local<F: Field, B: BitArray>(
     helper_role: Role,
     input: &[XorReplicated<B>],
+    num_bits: u32,
 ) -> Vec<Vec<BitConversionTriple<Replicated<F>>>> {
-    let mut total_list = Vec::new();
-    for bit_index in 0..B::BITS {
-        let one_list = input
-            .iter()
-            .map(|v| convert_bit_local::<F, B>(helper_role, bit_index, v))
-            .collect::<Vec<_>>();
-        total_list.push(one_list);
-    }
-    total_list
+    input
+        .iter()
+        .map(move |record| {
+            (0..num_bits)
+                .map(|bit_index: u32| {
+                    let left = u128::from(record.left()[bit_index]);
+                    let right = u128::from(record.right()[bit_index]);
+                    BitConversionTriple(match helper_role {
```

Comment:
Why not just invoke convert_bit_local?

Reply:
Missed it in multiple iterations. Changing it